src/lib/security/selinux_config.rb

```diff
@@ -0,0 +1,263 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "yast2/execute"
+
+module Security
+  # Class for handling SELinux kernel params
+  class SelinuxConfig
+    extend Yast::I18n
+    include Yast::I18n
+    include Yast::Logger
+
+    Yast.import "Bootloader"
+    Yast.import "ProductFeatures"
+
+    # @return [Security::SelinuxConfig::Mode] the last set mode, which can be differrent to the
+    #   {#running_mode} and {#configured_mode}. A call to {#save} is needed to make it the
+    #   {#configured_mode} for the next boot.
+    attr_reader :mode
+
+    # Constructor
+    def initialize
+      @mode = initial_mode
+    end
+
+    # Returns the initial SELinux mode
+    #
+    # It can be the proposed one if running in an installation or the configured mode for a
+    # running system
+    #
+    # @return [Security::SelinuxConfig::Mode]
+    def initial_mode
+      return @initial_mode if @initial_mode
+
+      propose! if Yast::Mode.installation
```

Comment:
result is not stored anywhere. I expect you want something like `@initial_mode = Yast::Mode.installation ? propose! : configured_mode`
and that reminds me why we need propose with side-efect? I do not like it much. Why not havve propose that just return proposed mode?

Reply:
maybe to be more consistent I would use `proposed_mode` as method name and result can be also nil if nothing should be proposed? not sure

Reply:
I thought that we needed to _store_ it as boot params to make the proposal effective. But, according to your concern, seem like it is not necessary. Honestly, I don't like side-effects neither. So, I'll happyly to getride of it and adapt this method too :)

Reply:
well, if we need it, we will use the save on place where it is needed. it will be move obvious.